lib/commands/connect/auth_connection.js

```diff
@@ -27,27 +29,24 @@ module.exports = {
     ],
     needsApp: true,
     needsAuth: true,
-    run: cli.command(function (context, heroku) {
-      return api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku)
-        .then(function(connections) {
-          if (connections.length > 0) {
-            callbackServer();
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withConnection(context, heroku);
+      callbackServer();
+
+      let url = '/api/v3/connections/' + connection.id + '/authorize_url';
```

Comment:
That's probably a good idea. For now, I just want to reproduce the existing functionality though, so I'll leave that cleanup for another PR.